torchinductor/codegen/triton.py

```diff
@@ -646,7 +653,7 @@ def indexing(self, index: sympy.Expr, copy_shape=None, dense_indexing=False):
                 mask.append(f"{tree.prefix}mask")
             dense_mask.append(f"{tree.prefix}mask")
 
-        if need_dense and not have_dense:
+        if (need_dense and not have_dense) or load_index0:
```

Comment:
Why not
```suggestion
        if (need_dense and not have_dense) or index == 0:
```
?

Reply:
I wasn't sure if this is safe to do if we are not calling from `load`, but CI run suggests this is ok.